### DISCUSSION TITLE
changing folder icons

### DISCUSSION THREAD

**Discussion by guillopacheco** [SylEleuth/gruvbox-plus-icon-pack#90] [Ideas]
Hello. I'm still tweaking with this icon pack, it's the most complete that I've seen! 24 folder color options! the only solution to make a faster color change for now in my little to none knowledge, is to open a terminal inside ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable and run these commands:
`ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato.svg folder.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-activities.svg folder-activities.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-apple.svg folder-apple.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-apps.svg folder-apps.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-atom.svg folder-atom.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-cd.svg folder-cd.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-cinnamon.svg folder-cinnamon.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-desktop.svg folder-desktop.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-development.svg folder-development.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-docker.svg folder-docker.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-documents.svg folder-documents.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-documents-open.svg folder-documents-open.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-download.svg folder-download.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-download-open.svg folder-download-open.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-downloads.svg folder-downloads.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-drag-accept.svg folder-drag-accept.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-dropbox.svg folder-dropbox.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-electron.svg folder-electron.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-favorites.svg folder-favorites.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-fonts.svg folder-fonts.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-games.svg folder-games.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-git.svg folder-git.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-github.svg folder-github.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-gitlab.svg folder-gitlab.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-gnome.svg folder-gnome.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-google-drive.svg folder-google-drive.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-icons.svg folder-icons.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-image-people.svg folder-image-people.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-important.svg folder-important.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-java.svg folder-java.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-linux.svg folder-linux.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-locked.svg folder-locked.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-mac.svg folder-mac.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-mail.svg folder-mail.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-mail-cloud.svg folder-mail-cloud.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-mega.svg folder-mega.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-meocloud.svg folder-meocloud.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-mozilla.svg folder-mozilla.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-music.svg folder-music.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-music-open.svg folder-music-open.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-network.svg folder-network.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-open.svg folder-open.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-owncloud.svg folder-owncloud.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-pcloud.svg folder-pcloud.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-photo.svg folder-photo.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-pictures.svg folder-pictures.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-pictures-open.svg folder-pictures-open.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-print.svg folder-print.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-private.svg folder-private.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-public.svg folder-public.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-publicshare-open.svg folder-publicshare-open.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-recent.svg folder-recent.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-remote.svg folder-remote.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-remote-open.svg folder-remote-open.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-saved-search.svg folder-saved-search.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-script.svg folder-script.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-steam.svg folder-steam.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-tar.svg folder-tar.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-templates.svg folder-templates.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-templates-open.svg folder-templates-open.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-themes.svg folder-themes.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-torrent.svg folder-torrent.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-unlocked.svg folder-unlocked.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-vbox.svg folder-vbox.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-video.svg folder-video.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-videos.svg folder-videos.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-videos-open.svg folder-videos-open.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-visiting.svg folder-visiting.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-vscode.svg folder-vscode.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-wifi.svg folder-wifi.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-wine.svg folder-wine.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato-yandex-disk.svg folder-yandex-disk.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/user-tomato-home.svg user-home.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/user-tomato-desktop.svg user-desktop.svg ;
ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/user-tomato-home-open.svg user-home-open.svg ;`

Want to change the color? copy the commands in a text editor, find and replace tomato for one of the other 23 options: red, rust, lime, jade, grey, gold, blue, white, olive, green, black, yellow, violet, tomato, purple, orange, citron, pumpkin, sapphire, lavender, highland, pistachio, firebrick, custom. save, copy and paste all commands in the terminal, run, logout, login. This is for my first issue

Second Issue, I'm unable to associate the 3d files (.stl .obj. .3mf) to the icon inside the icon pack. how can I do that? Thank you for your help and this awesome icon pack

**Comment by SylEleuth** [comment]
`ln -sfn ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable/folder-tomato.svg folder.svg` will create a new file with absolute path stated in the command. You shouldn't do that way and use only relative links. Also executing that command will create a new file in a directory You are at the moment of execution. That is why nothing is changing. The best way is to go to the places/scalable directly and use `ln -sfn folder-tomato.svg folder.svg` and so on. You could put it in the for loop and close it in few lines. I have plans to do a proper scripts. Unfortunately I don't know how to change relative link while not being within a folder. Maybe someone smarter than me could help with that.

I've been experimenting a little bit and going through all scalable icons it looks that'll need some work. There are color icons missing like bookmarks, trash and others. And linked files are scattered so much that I was lost after some time. So I will need to change some link paths to organize it better. Right now I'm hard working on the action icons and it will take some time until I jump to "places".

Also You don't have to logout after that. Just change icon pack to something different and go back to gruvbox. It works in KDE at least.

About 3d files could You open a new issue and explain what You exactly mean with examples?

**Comment by guillopacheco** [comment]
yes, the change works when executed inside the  ~/.local/share/icons/Gruvbox-Plus-Dark/places/scalable folder. but in gnome I have to logout and login to see the changes. thank you!
